pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -145,6 +147,34 @@ func (p *PodConverter) dataplaneFor(
 			dataplane.Networking.TransparentProxying.ReachableServices = reachableServicesValue
 			reachableServices = reachableServicesValue
 		}
+		if reachableBackendsRef, exist := annotations.GetString(metadata.KumaReachableBackendsRef); exist {
+			refs := ReachableBackendRefs{}
+			err := yaml.Unmarshal([]byte(reachableBackendsRef), &refs)
+			if err != nil {
+				return nil, errors.Errorf("cannot parse, %s has invalid format", metadata.KumaReachableBackendsRef)
+			}
+			backendRefs := []*mesh_proto.Dataplane_Networking_TransparentProxying_ReachableBackendRef{}
+			for _, ref := range refs.Refs {
+				backendRef := &mesh_proto.Dataplane_Networking_TransparentProxying_ReachableBackendRef{
+					Kind:   ref.Kind,
+					Labels: ref.Labels,
+				}
+				if ref.Port != nil {
+					backendRef.Port = util_proto.UInt32(pointer.Deref(ref.Port))
+				}
+				if ref.Name != nil {
+					backendRef.Name = pointer.Deref(ref.Name)
```

Comment:
pointer.Deref does not make sense when you already checked if pointer is not nil. Did you mean
```
backendRef.Name = pointer.Deref(ref.Name)
backendRef.Namespace = pointer.Deref(ref.Namespace)
```
? without ifs
or
```
backendRef.Name = *ref.Name
```
with ifs